### PR TITLE
Fix code style with eslint

### PR DESCRIPTION
**Pull request by jrobeson**
This setup is based off https://github.com/sidorares/node-mysql2

The main differences are the addition of `eslint:recommended` and these rules:

```json
  "rules": {
    "no-constant-condition": ["error", { "checkLoops": false }],
    "no-console": "off",
    "no-empty": "off"
  }
```

I don't see us ever removing the first, but hopefully we can eventually do something about second and third.

This PR originated from discussions in  #175

### CODE REVIEW

**Comment by jrobeson** [comment]
I punted on the variable shadowing for now.

**Comment by sidorares** [comment]
Node 4: "SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode" in /lib/message.js:43

Add "use strict" maybe?

**Comment by jrobeson** [comment]
If we need `'use strict';` , then we should probably enforce it on every file with eslint.

**Comment by jrobeson** [comment]
I should have phrased that as question. Should we enforce it with eslint?

**Comment by sidorares** [comment]
I wonder if `node --use_strict` is enough to solve node 4.x issue in tests -
- https://stackoverflow.com/questions/9031888/any-way-to-force-strict-mode-in-node

**Comment by jrobeson** [comment]
Is it worth it? isn't adding `'use strict'';` better if we still do support 4.x fully?

I personally don't use node 4.x anywhere, but it sounded like you really wanted to keep 4.x support. I assume due to LTS on ubuntu and debian?

**Comment by jrobeson** [comment]
Alternatively we could have a build step and transpile a bit, but that might be overkill (even though we'd get some nice features with it).

**Comment by sidorares** [comment]
I suggest simpler solution: not to use features that not supported by non-strict node 4 (in our case moving `let` one level upper from for loop)

**Comment by jrobeson** [comment]
I guess that means reverting https://github.com/sidorares/dbus-native/pull/182/commits/a9afe0804dbd19bd21f7d95a3911bcac8ef41447

**Comment by jrobeson** [comment]
I guess that means we can't export actual classes either. I think they require strict mode too then. I obviously didn't know how to read http://node.green/ properly

**Comment by jrobeson** [comment]
@sidorares : to be cleari: you can't just move let one level up. it's still block level scoped. it has to go back to being a var.

**Comment by jrobeson** [comment]
yeah. i'm going to have to switch everything but the top level consts and lets back to var. This was my mistake for not running all the work i did with: `nvm exec --lts=argon`

**Comment by jrobeson** [comment]
so another question. when previous code was like:

```js
  var a, b, c, d;

  // a bunch of code
  // ......
  for(a=0; ....) {
  }
```

and then i changed it to move `a` to the loop.

Should I change that back to the way it was where the variable hoisting was clear? Or should i just replace `let a = 0;` with `var a = 0;`

**Comment by jrobeson** [comment]
ah, there are enough where the variable is declared in the loop with var for me to go with that.

**Comment by jrobeson** [comment]
rebased. It should pass now.

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    "prettier"
+  ],
+  "parserOptions": {
+    "ecmaVersion": 6
+  },
+  "plugins": [
+    "markdown"
+  ],
+  "rules": {
+    "no-constant-condition": ["error", { "checkLoops": false }],
```
Comment (jrobeson) [inline .eslintrc]: I'm wondering if i should just inline overrides for this one where needed.

Comment (sidorares) [reply .eslintrc]: maybe. I'll merge what we have now and we'll iterate

**Comment by jrobeson** [comment]
so what next then?

**Comment by sidorares** [comment]
I think this is list of priorities:
1) better docs on mapping between js types anddbus types. Add common examples ( network manager, unity )
2) better docs on how to export object as dbus service

Potentially breaking changes: ( v2.0 ? )
1) make it much easier to map objects to variants and back ( things like 'a{sv}' )
2) Change from array-based method definition [ input signature, output signature, input description, output description] to named using object instead of array ( #170 )

**Comment by jrobeson** [comment]
I'd be hoping for a promise based API in 2.x if we're going to break BC. I see node-mysql2 has such a possibility and also the async keyword somehow.

**Comment by sidorares** [comment]
yes, promises are high on the priority list. I think we could even have that without braking current api

**Comment by sidorares** [comment]
node-mysql2 does this by exposing separate special wrapper, this is ok but not ideal

**Comment by jrobeson** [comment]
I wanna adjust the priority list a bit.  See if we can whittle down the PRs and take what's useful. I really can't do it well myself though.

**Comment by sidorares** [comment]
I'll try to help as much as I can but don't promise too much ( 6 kids and day job... )

**Pull request merged**